src/app/GitExtUtils/GitUI/ComboBoxExtensions.cs

```diff
@@ -21,37 +18,73 @@ public static void AdjustWidthToFitContent(this ComboBox comboBox)
 
         public static void ResizeDropDownWidth(this ComboBox comboBox, int minWidth, int maxWidth)
         {
-            if (comboBox is null)
-            {
-                throw new ArgumentNullException(nameof(comboBox));
-            }
+            ArgumentNullException.ThrowIfNull(comboBox);
 
             int calculatedWidth = GetPreferredDropDownWidth(comboBox);
             comboBox.DropDownWidth = Math.Min(Math.Max(calculatedWidth, minWidth), maxWidth);
         }
 
         public static void ResizeDropDownWidth(this ToolStripComboBox comboBox, int minWidth, int maxWidth)
         {
-            if (comboBox is null)
-            {
-                throw new ArgumentNullException(nameof(comboBox));
-            }
+            ArgumentNullException.ThrowIfNull(comboBox);
 
-            int calculatedWidth = GetPreferredDropDownWidth(comboBox.Control);
+            int calculatedWidth = GetPreferredDropDownWidth((ComboBox)comboBox.Control);
             comboBox.DropDownWidth = Math.Min(Math.Max(calculatedWidth, minWidth), maxWidth);
         }
 
-        private static int GetPreferredDropDownWidth(dynamic comboBox)
+        private static int GetPreferredDropDownWidth(ComboBox comboBox)
         {
             int calculatedWidth = 0;
-            using dynamic graphics = comboBox.CreateGraphics();
-            foreach (object obj in comboBox.Items)
+            using Graphics graphics = comboBox.CreateGraphics();
+
+            string displayMemberName = comboBox.DisplayMember;
+            Font font = comboBox.Font;
+
+            foreach (object item in comboBox.Items)
             {
-                dynamic area = graphics.MeasureString(obj.ToString(), comboBox.Font);
+                SizeF area = graphics.MeasureString(GetDisplayValue(displayMemberName, item), font);
                 calculatedWidth = Math.Max((int)area.Width, calculatedWidth);
             }
 
-            return calculatedWidth;
+            return calculatedWidth + CalculateVerticalScrollBarWidth(comboBox);
+        }
+
+        private static string? GetDisplayValue(string displayMemberName, object item)
+        {
+            if (displayMemberName.Length == 0)
+            {
+                return item.ToString();
+            }
+
+            System.Reflection.PropertyInfo? displayMemberProperty = item
+                .GetType()
+                .GetProperty(displayMemberName);
```

Comment:
We're doing this in a loop, which isn't great.

Reply:
I don't think this is an issue. You'd need ~55 items in a list for this to take `1ms`. Also it has to be in the loop as each item can be of a different type. We could use dictionary to cache these by type but IMO the code would get more confusing and complicated without perceivable benefit.

![image](https://github.com/user-attachments/assets/b35f7640-3096-445e-a198-8aed594d4a71)